src/psyclone/domain/nemo/transformations/nemo_arrayrange2loop_trans.py

```diff
@@ -308,16 +302,71 @@ def validate(self, node, options=None):
                 "supplied node argument should be within an ArrayReference "
                 "node that is within the left-hand-side of an Assignment "
                 "node, but it is on the right-hand-side.")
-        # Does the rhs of the assignment have any operations that
-        # return arrays as this is not currently supported?
+        # Does the rhs of the assignment have any operations that are not
+        # elemental?
         for operation in assignment.rhs.walk(Operation):
-            # At the moment the only array valued operator is matmul
-            if operation.operator == BinaryOperation.Operator.MATMUL:
+            # Allow non elemental UBOUND and LBOUND
+            if operation.operator is BinaryOperation.Operator.LBOUND:
+                continue
+            if operation.operator is BinaryOperation.Operator.UBOUND:
+                continue
+            if not operation.is_elemental():
                 raise TransformationError(
                     "Error in NemoArrayRange2LoopTrans transformation. This "
                     "transformation does not support array valued operations "
                     "on the rhs of the associated Assignment node, but found "
                     "'{0}'.".format(operation.operator.name))
+        # Do not allow to transform expressions with CodeBlocks
+        if assignment.walk(CodeBlock):
+            raise TransformationError(
+                "Error in NemoArrayRange2LoopTrans transformation. This "
```

Comment:
If possible, please could you include the Fortran for the problematic assignment in this error message?

Reply:
Yes, I use LazyString since this transformation is called often to find where is it applicable without needed the error message.

Reply:
Great. I thought one of the points of f-strings was that they automatically did lazy evaluation but I googled and I see that they don't.